File: client/src/app/detector/distro.py
import platform
from pathlib import Path


def detect_distro() -> str | None:
    if platform.system().lower() != "linux":
        return None

    os_release = Path("/etc/os-release")
    if not os_release.exists():
        return None

    data = {}
    for line in os_release.read_text().splitlines():
        if "=" in line:
            key, _, value = line.partition("=")
            data[key.strip()] = value.strip().strip('"')

    distro_id = data.get("ID", "").lower()

    match distro_id:
        case "arch" | "manjaro" | "endeavouros":
            return "arch"
        case "ubuntu" | "debian" | "linuxmint":
            return "debian"
        case "fedora" | "rhel" | "centos":
            return "fedora"
        case _:
            return distro_id or None

File: client/internal/detector/distro.go
package detector

import (
	"os"
	"strings"
)

func detectDistro() string {
	data, err := os.ReadFile("/etc/os-release")
	if err != nil {
		return ""
	}

	kv := make(map[string]string)
	for _, line := range strings.Split(string(data), "\n") {
		key, val, ok := strings.Cut(line, "=")
		if !ok {
			continue
		}
		kv[strings.TrimSpace(key)] = strings.Trim(strings.TrimSpace(val), "\"")
	}

	id := strings.ToLower(kv["ID"])

	switch id {
	case "arch", "manjaro", "endeavouros":
		return "arch"
	case "ubuntu", "debian", "linuxmint":
		return "debian"
	case "fedora", "rhel", "centos":
		return "fedora"
	default:
		return id
	}
}
